Mage.Sets/src/mage/cards/t/TimotharBaronOfBats.java

```diff
@@ -0,0 +1,139 @@
+import mage.players.Player;
+import mage.target.targetpointer.FixedTarget;
+
+import java.util.UUID;
+
+public class TimotharBaronOfBats extends CardImpl {
+
+    private static final FilterPermanent anotherVampireFilter = new FilterControlledCreaturePermanent("another nontoken Vampire you control");
+    static {
+        anotherVampireFilter.add(AnotherPredicate.instance);
+        anotherVampireFilter.add(SubType.VAMPIRE.getPredicate());
+    }
+
+    public TimotharBaronOfBats(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{4}{B}{B}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.addSubType(SubType.VAMPIRE);
+        this.addSubType(SubType.NOBLE);
+
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(4);
+
+        // Ward—Discard a card.
+        this.addAbility(new WardAbility(new DiscardCardCost()));
+
+        // Whenever another nontoken Vampire you control dies, you may pay {1} and exile it.
+        // If you do, create a 1/1 black Bat creature token with flying.
+        // It gains “When this creature deals combat damage to a player,
+        // sacrifice it and return the exiled card to the battlefield tapped.”
+        this.addAbility(new DiesCreatureTriggeredAbility(
+                new DoIfCostPaid(new TimotharBaronOfBatsCreateBatEffect(), new ManaCostsImpl<>("{1}"))
+                        .setText("you may pay {1} and exile it. " +
+                                "If you do, create a 1/1 black Bat creature token with flying. " +
+                                "It gains \"When this creature deals combat damage to a player, " +
+                                "sacrifice it and return the exiled card to the battlefield tapped\"."),
+                false,
+                anotherVampireFilter,
+                true
+        ));
+    }
+
+    private TimotharBaronOfBats(final TimotharBaronOfBats card) { super(card); }
+
+    @Override
+    public Card copy() { return new TimotharBaronOfBats(this); }
+}
+
+class TimotharBaronOfBatsCreateBatEffect extends OneShotEffect {
+
+    TimotharBaronOfBatsCreateBatEffect() {
+        super(Outcome.Benefit);
+        staticText = "exile it. " +
+                "If you do, create a 1/1 black Bat creature token with flying. " +
+                "It gains \"When this creature deals combat damage to a player, " +
+                "sacrifice it and return the exiled card to the battlefield tapped\".";
+    }
+
+    private TimotharBaronOfBatsCreateBatEffect(final TimotharBaronOfBatsCreateBatEffect effect) { super(effect); }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) { return false; }
+
+        Card card = game.getCard(targetPointer.getFirst(game, source));
+        if (card == null) { return false; }
+
+        player.moveCards(card, Zone.EXILED, source, game);
```

Comment:
Exiling the card is part of the cost "pay {1} and exile it".  I think this needs to be an `ExileTargetCost`.  Looks like `DoIfCostPaid` only works with a single cost.  I think it would need to be refactored to work with multiple costs for this to work correctly.

Reply:
Would it need multiple costs and refactoring, or could it work with a CompositeCost of the ManaCost and an appropriately-crafted Exile Cost? I briefly looked at how you'd implement this card because I wanted to put it into my Vampires deck.

Reply:
>Looks like DoIfCostPaid only works with a single cost

You can construct a CostsImpl object by hand and pass it as the cost argument to DoIfCostPaid. Look at Wolf of Devil's Breach for an example.

Reply:
^^
Or do something like those above comments.  I haven't looked into it much but I'm pretty sure it needs to be a cost rather than part of the effect.

Reply:
So I see there's an ExileFromGraveyardCost(). But does the card hit the graveyard with Timothar, or do you exile it instead of putting it into the graveyard?

Reply:
Oh yea, my bad it would hit the graveyard first (it's a dies trigger) and then get exiled from the graveyard if the player chooses to pay the cost.

Reply:
Oh good, they I can use the existing cost rather than have to make my own.

Reply:
Is there a way for my to specify the Target for ExileTargetCost while I'm constructing the cost? Since I don't know it ahead of time.

Reply:
I think you would need to use a CostAdjuster.

Reply:
I've half implemented that. I am just not sure how to obtain access to the card which set off the DiesCreatureTriggeredAbility. I'll look into it some more, but if anyone knows off the top of their head, it would be much appreciated.

Reply:
`ability.getEffects().get(0).getTargetPointer()` should work I think

Reply:
So the newest push has the ExileCard cost. But I'm doing something wrong. It exiles the card before the DoIfPaid prompt shows up, and does not create a bat.

Reply:
Yea, don't use `addTarget`.  I think the best way is to create a `FilterCard` with a `CardIdPredicate`.

Reply:
That's not better. The exile still happens before the DoIfCostPaid popup, but now the player has to manually exile the card. Also, the bat still doesn't get made.

I don't think I have the cost adjuster in the right place, but I can't figure out where else it would go.

Reply:
Oh, yea you're adding the cost to the ability itself.  It needs to be added to the DoIfCostPaid effect's costs.  Maybe CostAdjuster isn't the right way to go.  Kind of unsure on this one to be honest.

Reply:
Probably not worth it to jump though all these hoops just to use `DoIfCostPaid`, you can just write a custom effect that handles the payment and creating the token. 

Also you should give it the ability with `GainAbilityTargetEffect` instead of adding it to the token itself, you can use the `FixedTargets` constructor that accepts a token as an argument

Reply:
I will try out the custom effect suggestion.

Out of curiosity, why the `GainAbilityTaretEffect`?

Reply:
it says that it “gains” the ability, which means if you make a copy of the token the copy won’t have the gained ability as it’s not a copiable value

Reply:
Looking at the ruling on Timothar, it looks somewhat in-between a full copy of the ability, and no copy of the ability:

"If another creature enters the battlefield as a copy of the Bat token or another effect creates a copy of the Bat token, it will have the triggered ability granted by Timothar, but will not be linked to Timothar’s ability and will not return the card to the battlefield. The triggered ability it gains, however, will still cause you to sacrifice the copy of the Bat when it deals combat damage to a player."

Reply:
hmmm, that seems contradictory but I’m gonna ask the rules manager about that

Reply:
These are the rules that drive everyone nuts.  What a weird interaction.

Reply:
![image](https://user-images.githubusercontent.com/29759218/152601971-6b381434-90c2-4940-a25c-5c788ce72c96.png)

looks like I was right!

Reply:
Phew. That's a lot less work.

Reply:
Okay, how's that